src/installer/installer.go

```diff
@@ -782,7 +784,7 @@ func (i *installer) updateReadyMasters(nodes *v1.NodeList, readyMasters *[]strin
 			if ok && (host.Host.Status == nil || *host.Host.Status != models.HostStatusInstalled) {
 				if err := i.inventoryClient.UpdateHostInstallProgress(ctx, host.Host.InfraEnvID.String(), host.Host.ID.String(), models.HostStageJoined, ""); err != nil {
 					log.Errorf("Failed to update node installation status, %s", err)
-					continue
+					return err
 				}
 			}
 			*readyMasters = append(*readyMasters, node.Name)
```

Comment:
@tsorya I guess you were right about https://github.com/openshift/assisted-installer/pull/823/files#r1566030218 - leaving that as it was would have avoided this problem. I didn't realise that the `inventoryHostsMap` here was always stale even on the retry.